src/init.cpp

```diff
@@ -1843,6 +1846,7 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
     connOptions.m_peer_connect_timeout = peer_connect_timeout;
     connOptions.whitelist_forcerelay = args.GetBoolArg("-whitelistforcerelay", DEFAULT_WHITELISTFORCERELAY);
     connOptions.whitelist_relay = args.GetBoolArg("-whitelistrelay", DEFAULT_WHITELISTRELAY);
+    connOptions.disable_v1conn_clearnet = args.GetBoolArg("-v2transport", DEFAULT_V2_TRANSPORT) && args.GetBoolArg("-v2onlyclearnet", false);
```

Comment:
```suggestion
    connOptions.disable_v1conn_clearnet = args.GetBoolArg("-v2onlyclearnet", false);
```

The InitError above means this should be safe